src/server.ts

```diff
@@ -88,14 +89,15 @@ app.post('/webhook', async (req, res) => {
       console.log(`Responding with Signature: ${signature}`)
       res.status(200).send(signature)
     } else if (req.body?.method === 'signTypedData') {
-      const { domain, types, value, address } = req.body
+      const { data, address } = req.body
 
       console.log(`SignTypedData request from address: ${address}`)
-      const signingKey = await walletService.getSigningKey(address)
-      const signature = signingKey.signDigest(
-        _TypedDataEncoder.hash(domain, types, value)
-      )
+      const privateKey = await walletService.getPrivateKey(address)
 
+      const jsonData = JSON.parse(data)
+      const signature = signTypedData_v4(Buffer.from(privateKey.slice(2), "hex"), {
```

Comment:
worth noting that if the address was stored without the `0x` preview than the slice wouldn't be needed. In this case because portalex is example code, I don't think we need to add a check. but just wanted to add that note